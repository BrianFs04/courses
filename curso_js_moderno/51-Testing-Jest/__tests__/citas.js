import Citas from "../js/classes/Citas";

describe("Probar la clase de citas", () => {
    const citas = new Citas();
    const id = Date.now();
    test("Agregar una nueva cita", () => {
        const citaObj = {
            id,
            mascota: "Hoop",
            propietario: "Brian",
            telefono: "123123123",
            fecha: "10-12-2020",
            hora: "10:30",
            sintomas: "Solo duerme",
        };

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test("Actualizar cita", () => {
        const citaActualizada = {
            id,
            mascota: "New Name",
            propietario: "Brian",
            telefono: "123123123",
            fecha: "10-12-2020",
            hora: "10:30",
            sintomas: "Solo duerme",
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test("Eliminar cita", () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});
