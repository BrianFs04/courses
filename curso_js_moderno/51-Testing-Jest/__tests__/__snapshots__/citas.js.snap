// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1612369489642,
      "mascota": "New Name",
      "propietario": "Brian",
      "sintomas": "Solo duerme",
      "telefono": "123123123",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1612369489642,
      "mascota": "Hoop",
      "propietario": "Brian",
      "sintomas": "Solo duerme",
      "telefono": "123123123",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
